kore.maude

```diff
@@ -220,18 +227,30 @@ Questions about this definition:
 - do we really need ~> as a K construct?  Can it be a label and thus the way K
   deal with evaluation contexts just a particular methodology and not an
   intrisic part of K ? Same question for =>.
+  Cosmin: Syntacticly at least, I would say "no" for both. From the implementation
+  perspective, this will simplify traversal code even further. 
 - rename .::K and .::KList to . ?  if we get rid of ~> as above, then we may
   only need .::KList.
+  Cosmin: I agree.
 - Should we allow empty productions instead of sort declarations?  A pro
   argument is that we have fewer syntactic constructs.  A cons argument is
   that a sort declaration may be regarded as a different entityr, with
   different kinds of attributes.
+  Cosmin: I don't quite follow.
 - Should we add constant AttributeKeys which have special semantics, like
   "strict", as constants?
+  Cosmin: we may want to discuss the attribute syntax separately.
 - Brackets and disappearing injections are the same thing, arent's they?  A
   production which only has one non-terminal and the production disappears
   from the AST.
+  Cosmin: Yes, I also think they can be seen as the same thing. That being
+          said, I would really like for injections to not dissapear. If something
+          is meant to dissapear, it should be declared as a subsort. And the
+          dissapearing injections (including subsorts) could be done via rules as
+          we discussed before.
```

Comment:
No.  I think by default all productions should have unique labels associated to them.  If you do not want a label, then you can put some attribute, like bracket or subsort, which will generate a rule that removes the label.  I am not concerned about attaching semantics to labels, because we already do it anyway with several of them, and so does maude, obj, cafeobj, etc.



Sent from my Verizon Wireless 4G LTE smartphone


-------- Original message --------
From: Cosmin Radoi <notifications@github.com>
Date: 04/07/2016 08:00 (GMT+01:00)
To: grosu/kore-in-maude <kore-in-maude@noreply.github.com>
Cc: "Rosu, Grigore" <grosu@illinois.edu>
Subject: Re: [grosu/kore-in-maude] discussion (#1)


In kore.maude<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_grosu_kore-2Din-2Dmaude_pull_1-23discussion-5Fr58822475&d=BQMCaQ&c=8hUWFZcy2Z-Za5rBPlktOQ&r=gBub4ab7wvmjFVEDwsswdud0fptTHrGi3G4Zzc5Kwds&m=tMtUt_dmUvCV0T_JOIsgBhmzc6E1OutCg6ZyD-urFck&s=qKJvEl_2QoXxWYYUzeRrKbQiCLmlA7hIdeNiJwscfbo&e=>:

>  - Brackets and disappearing injections are the same thing, arent's they?  A
>    production which only has one non-terminal and the production disappears
>    from the AST.
> +  Cosmin: Yes, I also think they can be seen as the same thing. That being
> +          said, I would really like for injections to not dissapear. If something
> +          is meant to dissapear, it should be declared as a subsort. And the
> +          dissapearing injections (including subsorts) could be done via rules as
> +          we discussed before.


Ok. The question is: is the injection klabel from syntax A ::= B (written exactly like this, with no attributes) automatically removed?

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_grosu_kore-2Din-2Dmaude_pull_1_files_c30c2bc78e45acdfc81ba5555f94fbe58d537506-23r58822475&d=BQMCaQ&c=8hUWFZcy2Z-Za5rBPlktOQ&r=gBub4ab7wvmjFVEDwsswdud0fptTHrGi3G4Zzc5Kwds&m=tMtUt_dmUvCV0T_JOIsgBhmzc6E1OutCg6ZyD-urFck&s=6H_YP_qGNYMqYS-rBZYqnKWwtOK_lzKSqL2BFTMWSD8&e=>